assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -312,6 +338,71 @@ const SelectedVehicleDataLayers = ({
   )
 }
 
+const SelectedRouteDataLayers = ({
+  routePatternIdentifier,
+  routePatterns,
+  showSelectionCard,
+  deleteSelection,
+  selectVehicle,
+  selectRoutePattern,
+  setStateClasses,
+}: {
+  routePatternIdentifier: RoutePatternIdentifier
+  routePatterns: ByRoutePatternId<RoutePattern> | null
+  showSelectionCard: boolean
+  selectVehicle: (vehicleOrGhost: VehicleOrGhost) => void
+  selectRoutePattern: (routePattern: RoutePattern) => void
+  deleteSelection: () => void
+
+  setStateClasses: (classes: string | undefined) => void
+}) => {
+  const selectedRoutePattern: RoutePattern | undefined = routePatterns
+    ? routePatterns[routePatternIdentifier.routePatternId]
+    : undefined
+  const position = selectedRoutePattern
+    ? selectedRoutePattern.shape?.points?.map((p) =>
+        Leaflet.latLng(p.lat, p.lon)
+      ) || []
+    : []
+  const followerState = useInteractiveFollowerState()
+
+  useEffect(() => {
+    setStateClasses(FollowerStatusClasses(followerState.shouldFollow))
+  }, [followerState.shouldFollow, setStateClasses])
```

Comment:
**suggestion:** I'm not actually sure that you need a `useEffect` here in order to prevent excessive re-renders: according to the documentation, [state updates are ignored if the new value equals the old](https://beta.reactjs.org/reference/react/useState#ive-updated-the-state-but-the-screen-doesnt-update).

Reply:
It seems that the `useEffect` is needed here, not for preventing excessive re-renders, but rather for performing the state update of `stateClasses` as a side effect. When removing this, I get the following errors in tests:
```
console.error
      Warning: Cannot update a component (`MapDisplay`) while rendering a different component (`SelectedRouteDataLayers`). To locate the bad setState() call inside `SelectedRouteDataLayers`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render

```

Reply:
Ahh, okay, my bad.